solr/core/src/test/org/apache/solr/search/TestExtendedDismaxParser.java

```diff
@@ -1771,6 +1775,35 @@ public void testSplitOnWhitespace_Basic() throws Exception {
     assertThat(parsedquery, anyOf(containsString("((name:stigma | title:stigma))"), containsString("((title:stigma | name:stigma))")));
   }
 
+    @Test 
+    public void testQf_stringFieldWithSowFalse_shouldBuildSingleQueryClause() throws Exception
+    {
+        assertJQ(req("qf", "trait_ss", "defType", "edismax", "q", "multi term", "sow", "false"),
+            "/response/numFound==1", "/response/docs/[0]/id=='75'");
+
+        String parsedquery = getParsedQuery(
+            req("qf", "trait_ss", "q", "multi term", "defType", "edismax", "sow", "false", "debugQuery", "true"));
+        assertThat(parsedquery, anyOf(containsString("((trait_ss:multi term))")));
+    }
+
+    @Test
+    public void testQf_numericField_shouldBuildMultipleQueryClausesIndependentlyOfSow() throws Exception
+    {
+        assertJQ(req("qf", "foo_i", "defType", "edismax", "q", "101 102", "sow", "false"),
+            "/response/numFound==2", "/response/docs/[0]/id=='76'", "/response/docs/[1]/id=='77'");
+
+        String parsedquery = getParsedQuery(
+            req("qf", "foo_i", "q", "101 102", "defType", "edismax", "sow", "false", "debugQuery", "true"));
+        assertThat(parsedquery, anyOf(containsString("foo_i:[101 TO 101]"), containsString("foo_i:[102 TO 102]")));
```

Comment:
I think you may want `allOf(...)` here (and line 1804 below) instead of `anyOf(...)`

Reply:
you are right, fixed!